services/galley/test/integration/API/Federation.hs

```diff
@@ -161,15 +162,16 @@ addLocalUser = do
           { FedGalley.cuTime = now,
             FedGalley.cuOrigUserId = qbob,
             FedGalley.cuConvId = conv,
-            FedGalley.cuAlreadyPresentUsers = [],
+            FedGalley.cuAlreadyPresentUsers = [charlie],
             FedGalley.cuAction =
               ConversationActionAddMembers (pure (qalice, roleNameWireMember))
           }
-  WS.bracketR c alice $ \ws -> do
+  WS.bracketR2 c alice charlie $ \(wsA, wsC) -> do
     FedGalley.onConversationUpdated fedGalleyClient remoteDomain cu
-    void . liftIO $
-      WS.assertMatch (5 # Second) ws $
+    liftIO $ do
+      WS.assertMatch_ (5 # Second) wsA $
         wsAssertMemberJoinWithRole qconv qbob [qalice] roleNameWireMember
+      WS.assertNoEvent (1 # Second) [wsC]
```

Comment:
The general principle at play here is that, although we normally trust the conversation-owning backend about the state of a conversation, we also try to protect our local users from spam. So if Charlie is not in the conversation, according to our db, then we cannot know at this point whether it is safe to notify him or not. So we conservatively omit the notification, and leave it to some future "reconciliation" procedure to forward the missing notification, in case we find out he was really there after all.

Note that at this point this doesn't really offer any protection, because we are not checking connections when a user is added. But that will change soon, hopefully.